File: internal/zinc-compile-core/src/main/scala/sbt/internal/inc/NoopClassFileManager.scala
package sbt.internal.inc

import java.io.File

import xsbti.compile.ClassFileManager

/**
 * A noop ClassFileManager that applies to non-javac tools.
 */
class NoopClassFileManager extends ClassFileManager {
  override def delete(classes: Array[File]): Unit = ()
  override def generated(classes: Array[File]): Unit = ()
  override def useCustomizedFileManager(): Boolean = false
  override def complete(success: Boolean): Unit = ()
}

File: internal/compiler-interface/src/main/java/xsbti/compile/NoopClassFileManager.java
package xsbti.compile;

import java.io.File;

public class NoopClassFileManager implements ClassFileManager {
    @Override
    public void delete(File[] classes) {
    }

    @Override
    public void generated(File[] classes) {
    }

    @Override
    public void complete(boolean success) {
    }
}
